app/models/sitemap.rb

```diff
@@ -0,0 +1,107 @@
+require "builder"
+
+class Sitemap
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  include Sitemap::RobotsUpdatable
+
+  ERROR_PAGES = %w[400.html 404.html 406-unsupported-browser.html 422.html 500.html].freeze
+
+  attribute :base_url, :string
+  attribute :build_dir, :string, default: -> { Parklife.application.config.build_dir }
+  attribute :update_robots, :boolean, default: false
+
+  validates :base_url, presence: true, format: { with: URI.regexp(%w[http https]) }
+
+  def generate!
+    write_sitemap!
+    write_compressed_sitemap!
+
+    handle_robots_update!
+
+    total_pages
+  end
+
+  private
+
+  # File writing
+  def write_sitemap!
+    File.write(sitemap_path, xml_sitemap)
+
```

Comment:
The `puts` statements mix business logic with presentation concerns. Consider using a logging mechanism or moving output messages to the caller for better separation of concerns.